Userland/Libraries/LibWeb/DOMURL/DOMURL.cpp

```diff
@@ -10,27 +10,27 @@
 #include <AK/IPv6Address.h>
 #include <AK/URLParser.h>
 #include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/DOMURL/DOMURL.h>
 #include <LibWeb/FileAPI/Blob.h>
 #include <LibWeb/FileAPI/BlobURLStore.h>
-#include <LibWeb/URL/URL.h>
 
-namespace Web::URL {
+namespace Web::DOMURL {
 
-JS_DEFINE_ALLOCATOR(URL);
+JS_DEFINE_ALLOCATOR(DOMURL);
 
-JS::NonnullGCPtr<URL> URL::create(JS::Realm& realm, AK::URL url, JS::NonnullGCPtr<URLSearchParams> query)
+JS::NonnullGCPtr<DOMURL> DOMURL::create(JS::Realm& realm, URL url, JS::NonnullGCPtr<URLSearchParams> query)
```

Comment:
I've been hemming and hawing over what a name for this should be, especially since it's a bit awkward with the namespace as well. Eventually I decided on consulting other engines on what they do - since I figured that may have the exact same problem, and noticed that they actually used DOMURL.

Open to other ideas on this, I'm not fully convinced

Reply:
A long while back in discord `URLObject` was suggested - maybe that's better? But the namespace is a bit awkward (though I'm not sure on DOMURL either to be fair)